nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizer.java

```diff
@@ -667,6 +682,8 @@ private Set<String> updateConnectionDestinations(final ProcessGroup group, final
                 newDestination == null
                 ||
                 (newDestination.getConnectableType() == ConnectableType.OUTPUT_PORT && !newDestination.getProcessGroup().equals(connection.getProcessGroup()))
+                ||
+                (newDestination.getConnectableType() == ConnectableType.INPUT_PORT && !connection.getDestination().getVersionedComponentId().get().equals(proposedConnection.getDestination().getId()) && connection.getDestination().getName().equals(proposedConnection.getDestination().getName()))
```

Comment:
I could be wrong, but I'm pretty sure that this block of code is not actually necessary. We don't need to handle this case specially, as far as I can tell. Unless there's something that I'm missing.